### PR TITLE
added send-and-confirm functions for those expecting SendableDataTransaction

### PR DESCRIPTION
**Pull request by DiosAkhilleus**

### CODE REVIEW

File: solana/gatekeeper-lib/test/GatekeeperService.test.ts [hunk 1]
```diff
@@ -139,6 +150,39 @@ describe("GatekeeperService", () => {
...
 
+  context("sendIssue", () => {
+    beforeEach(() => {
+      sandbox
+        .stub(GatewayTs, "getGatewayTokenAddressForOwnerAndGatekeeperNetwork")
+        .resolves(gatewayTokenAddress);
+      sandbox
+        .stub(GatewayTs, "getGatekeeperAccountAddress")
+        .resolves(gatekeeperAccountAddress);
+      sandbox.stub(GatewayTs, "getGatewayToken").resolves(activeGatewayToken);
```
Comment (DiosAkhilleus) [inline solana/gatekeeper-lib/test/GatekeeperService.test.ts:161]: The stubbing does not appear to function correctly

File: solana/gatekeeper-lib/src/util/connection.ts [hunk 2]
```diff
@@ -102,41 +102,11 @@ export class SendableTransaction implements TransactionHolder {
...
-    return {
-      message: message.serialize(),
-      signatures,
-    };
-  }
-
   static fromSerialized(
     connection: Connection,
-    message: Buffer,
-    signatures: string[]
```
Comment (rado0x54) [inline solana/gatekeeper-lib/src/util/connection.ts:134]: Missing context, but why was `signatures` removed here?

File: solana/gateway-ts/src/types/index.ts [hunk 3]
```diff
@@ -27,6 +27,18 @@ export class GatewayToken {
     return this.state === State.ACTIVE && !this.hasExpired();
   }
 
+  equals(other: GatewayToken): boolean {
```
Comment (rado0x54) [inline solana/gateway-ts/src/types/index.ts:30]: This is a very OOO way of programming and I'm not sure we need it anywhere APART from tests. Therefore I'd propose to remove it. (or move it into a test utility)

Why would a library consumer ever use that equals? (e.g. either he looks at the SAME token at the same address (and only compare that) OR it is very unlikely that expireTime ever matches.

**Review by rado0x54 — CHANGES_REQUESTED** [review]
Some minor comments.

**Review by Buzzec — CHANGES_REQUESTED** [review]

File: solana/gatekeeper-lib/src/service/GatekeeperNetworkService.ts [hunk 4]
```diff
@@ -16,6 +16,7 @@ import {
   TransactionOptions,
 } from "../util/transaction";
 import { SOLANA_COMMITMENT } from "../util/constants";
+import { Hash } from "crypto";
```
[the remark this answers is not in the document]
Comment (DiosAkhilleus) [reply solana/gatekeeper-lib/src/service/GatekeeperNetworkService.ts:19]: I don't think so. It must have auto-added the import while I was adding the new functions. I will remove it.

File: solana/gatekeeper-lib/src/util/connection.ts (hunk 2 above)
Comment (Buzzec) [reply solana/gatekeeper-lib/src/util/connection.ts:134]: The signatures are now in the buffer, it's the solana way of doing it where the whole transaction is serialized

File: solana/gatekeeper-lib/src/service/GatekeeperService.ts [hunk 5]
```diff
@@ -301,3 +299,74 @@ export class GatekeeperService {
...
+export class SimpleGatekeeperService {
+  gs: GatekeeperService;
+
+  /**
+   *
+   * @param connection
+   * @param gatekeeperNetwork
+   * @param gatekeeperAuthority
+   * @param config
+   */
```
[the remark this answers is not in the document]
Comment (DiosAkhilleus) [reply solana/gatekeeper-lib/src/service/GatekeeperService.ts:312]: Will do

File: solana/gatekeeper-lib/src/service/GatekeeperNetworkService.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (DiosAkhilleus) [reply solana/gatekeeper-lib/src/service/GatekeeperNetworkService.ts:19]: Removed

File: solana/gatekeeper-lib/src/service/GatekeeperService.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (DiosAkhilleus) [reply solana/gatekeeper-lib/src/service/GatekeeperService.ts:312]: Done

File: solana/gateway-ts/src/types/index.ts (hunk 3 above)
Comment (DiosAkhilleus) [reply solana/gateway-ts/src/types/index.ts]: Workaround for Chai's to.equal method. No longer necessary after updating test file

**Review by rado0x54 — APPROVED** [review]

**Pull request merged**